### ISSUE TITLE
Prepare F77 Version of the Subroutines and Example Programs

### ISSUE THREAD

**Issue by hjozwiak-umk** [nikodem-stolarczyk/spectral-line-shapes#52]
The goal of this task is to prepare Fortran 77 (F77) versions of the existing Fortran 90 (F90) subroutines and example programs in the project. Below are the key areas that need attention:

1. General Code Adjustments
   - Remove all F90 features unsupported in F77:
     - `module` and `use` statements
     - `kind` parameter syntax for precision
     - `optional` arguments and the `present()` intrinsic function
2. Subroutine and Function Adjustments
   - Replace `module` encapsulation with `include` for shared constants
   - Replace `optional` arguments with explicitly passed default values for previously optional arguments
   - Refactor `functions` with `result` syntax
3. Constants and Parameters
   - Replace `parameter` constants in modules with `include` files (`constants.inc`)
 4. I/O Adjustments
   - Rewrite `write` statements with long format strings using continuation lines or `format` labels
5. Example programs
   - Refactor `use` statements to directly reference subroutines with explicit declarations; explicitly declare return type of functions.
6. Update Makefile to handle `.f` files.

**Comment by SuoWick** [comment]
About optional parameters in F77: What about creating different "ENTRY" entries for the function depending on the input variable count? Function deprecated in releases newer than F77.

**Comment by hjozwiak-umk** [comment]
> About optional parameters in F77: What about creating different "ENTRY" entries for the function depending on the input variable count? Function deprecated in releases newer than F77.

Thanks for the suggestion, Michał. While `ENTRY` can indeed provide alternative entry points to a subroutine or function, it becomes impractical for a case like ours with multiple optional parameters. We currently have 4 optional arguments (`Xlm`, `Ylm`, `alpha`, `calculate_dispersion`), which means we’d need to create 16 different ENTRY points to account for all possible combinations of arguments being passed or omitted. For example:

- Only `Xlm` passed. 
- Both `Xlm` and `Ylm` passed.
- Only `alpha` passed.
- `calculate_dispersion` and `alpha` passed without line mixing coefficients.

This makes the code difficult to read, maintain, and debug. Each ENTRY point would require duplicating or partially duplicating the logic for setting defaults and handling arguments.

**Comment by SuoWick** [comment]
Yes and no. You can create limited number of cases to maintain "compatibility" with different languages (MATLAB for instance can only include alpha if you include Ylm and Xlm before). It would not require duplicating logic much as well as it might as well just call the "main" function with some of the parameters fixed to default value. However, I agree that this introduces a lot of work that will not contribute much to the project due to the scale of use of the F77. I'd say it's your call, just wanted to emphasise it is not impossible to reintroduce optional parameters here.

**Comment by hjozwiak-umk** [comment]
**Replace `real*8 `with `double precision`**

As pointed out by our colleagues that tested the Fortran77 implementation, `real*8` declaration may not hold for all compilers and architectures. `double precision` is part of the Fortran standard and ensures better portability and compiler support.

**Comment by hjozwiak-umk** [comment]
> **Replace `real*8 `with `double precision`**
> 
> As pointed out by our colleagues that tested the Fortran77 implementation, `real*8` declaration may not hold for all compilers and architectures. `double precision` is part of the Fortran standard and ensures better portability and compiler support.

See #56
